# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/meta/webcodecs/videoDecoder-codec-specific.https.any.js.ini
# Commit: 02cb31bfdc0f
# Full Hash: 02cb31bfdc0f4a15397b3a4e0725c408d738b34c
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Implement VideoDecoder::IsConfigSupported r=padenot
#   
#   This patch implements the isConfigSupported method for WebCodecs'
#   VideoDecoder.
#   
# ==============================================================================

diff -r 77b0f5226ed8 -r 02cb31bfdc0f testing/web-platform/meta/webcodecs/videoDecoder-codec-specific.https.any.js.ini
--- a/testing/web-platform/meta/webcodecs/videoDecoder-codec-specific.https.any.js.ini	Thu Jun 29 22:33:11 2023 +0000
+++ b/testing/web-platform/meta/webcodecs/videoDecoder-codec-specific.https.any.js.ini	Thu Jun 29 22:33:11 2023 +0000
@@ -1,14 +1,7 @@
 [videoDecoder-codec-specific.https.any.worker.html?vp8]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
@@ -57,16 +50,9 @@
 
 
 [videoDecoder-codec-specific.https.any.worker.html?h264_avc]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
@@ -115,16 +101,9 @@
 
 
 [videoDecoder-codec-specific.https.any.worker.html?vp9]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
@@ -173,16 +152,9 @@
 
 
 [videoDecoder-codec-specific.https.any.worker.html?h264_annexb]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
@@ -231,16 +203,9 @@
 
 
 [videoDecoder-codec-specific.https.any.html?vp8]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
@@ -289,16 +254,9 @@
 
 
 [videoDecoder-codec-specific.https.any.html?av1]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
@@ -347,16 +305,9 @@
 
 
 [videoDecoder-codec-specific.https.any.worker.html?av1]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
@@ -405,16 +356,9 @@
 
 
 [videoDecoder-codec-specific.https.any.html?vp9]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
@@ -463,16 +407,9 @@
 
 
 [videoDecoder-codec-specific.https.any.html?h264_avc]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
@@ -521,16 +458,9 @@
 
 
 [videoDecoder-codec-specific.https.any.html?h264_annexb]
+  prefs: [dom.media.webcodecs.enabled:true]
   expected:
     if (os == "android") and fission: [ERROR, TIMEOUT]
-  [Test isConfigSupported()]
-    expected: FAIL
-
-  [Test isConfigSupported() with 1080p crop]
-    expected: FAIL
-
-  [Test that isConfigSupported() returns a parsed configuration]
-    expected: FAIL
 
   [Test invalid configs]
     expected: FAIL
